src/main/java/com/zaxxer/nuprocess/internal/BasePosixProcess.java

```diff
@@ -317,6 +369,62 @@ else if (WIFSIGNALED(status)) {
       return this;
    }
 
+   private int spawnWithCwdSetViaPthreadChdir(
+      IntByReference restrict_pid,
+      String restrict_path,
+      Pointer file_actions,
+      Pointer /*const posix_spawnattr_t*/restrict_attrp,
+      StringArray /*String[]*/argv,
+      Pointer /*String[]*/envp,
+      Path cwd)
+   {
+     int cwdBufSize = 1024;
+     long peer = Native.malloc(cwdBufSize);
+     Pointer oldCwd = new Pointer(peer);
+     LibC.getcwd(oldCwd, cwdBufSize);
+     String newCwd = cwd.toAbsolutePath().toString();
+     int rc = LibC.SYSCALL.syscall(LibC.SYSCALL.SYS___pthread_chdir, newCwd);
+     checkReturnCode(rc, "syscall(SYS__pthread_chdir) failed to set current directory");
+
+     try {
+       return LibC.posix_spawnp(restrict_pid, restrict_path, file_actions, restrict_attrp, argv, envp);
+     } finally {
+       rc = LibC.SYSCALL.syscall(LibC.SYSCALL.SYS___pthread_chdir, oldCwd);
+       Native.free(Pointer.nativeValue(oldCwd));
+       checkReturnCode(rc, "syscall(SYS__pthread_chdir) failed to restore current directory");
+     }
+   }
+
+   private int spawnWithCwdSetViaLinuxCwdChangeableExecutorService(
+      final IntByReference restrict_pid,
+      final String restrict_path,
+      final Pointer file_actions,
+      final Pointer /*const posix_spawnattr_t*/restrict_attrp,
+      final StringArray /*String[]*/argv,
+      final Pointer /*String[]*/envp,
+      final Path cwd)
+   {
+     Future<Integer> setCwdThenSpawnFuture = linuxCwdChangeableExecutorService.submit(
+         new Callable<Integer>() {
+           @Override
+           public Integer call() {
+               // Set cwd in this thread, which has its cwd state disassociated from the rest of the process.
+               int rc = LibC.chdir(cwd.toAbsolutePath().toString());
+               checkReturnCode(rc, "chdir() failed");
```

Comment:
The `RuntimeException` is caught then wrapped in an `ExecutionException` which is re-thrown when we call `Future.get()`.

Added a comment and updated to explicitly handle and unwrap `ExecutionException` thrown by `Future.get()`.